java/piranha/src/main/java/com/uber/piranha/XPFlagCleaner.java

```diff
@@ -699,7 +699,19 @@ public Description matchBinary(BinaryTree tree, VisitorState state) {
     if (deletedSubTree != null) {
       Description.Builder builder = buildDescription(tree);
       SuggestedFix.Builder fixBuilder = SuggestedFix.builder();
-      fixBuilder.replace(tree, remainingSubTree.toString());
```

Comment:
If the tree kind is `ParenthesizedTree` here, then this utility [call](https://github.com/google/error-prone/blob/master/check_api/src/main/java/com/google/errorprone/util/ASTHelpers.java#L342) can be used.

Reply:
the tree kind is `conditional_and` in such cases. 
`parenthesised tree` is a couple levels above from the current node position

Reply:
Ok. Since there is a check for lone condition by getting the leftOperand, presumably, the parent can be checked for that type to refactor appropriately.

Reply:
since the tree does not have any back pointer to the parent node and these calls happen as the AST node visits happen we cannot directly use this particular method.
If we really want to use this method, we need to somehow get the parent tree pointer. And i guess this can be achieved using recursion based handling.

Reply:
FYI `state.getPath()` gives you the `TreePath` to the current AST node, from which you can retrieve all the ancestor nodes